src/rsa/signing.rs

```diff
@@ -176,181 +201,194 @@ impl KeyPair {
                 let dP = der::positive_integer(input)?;
                 let dQ = der::positive_integer(input)?;
                 let qInv = der::positive_integer(input)?;
```

Comment:
Above, we shouldn't use `der::positive_integer` for the parsing, because the checking for a positive number should be done in `from_key_input()`.